### ISSUE TITLE
Train Mask RCNN on custom dataset - rpn bbox loss doesn't get lower than 2

### ISSUE THREAD

**Issue by SebastianNagy** [matterport/Mask_RCNN#2577]
Hello everyone,

I'm trying to train the matterport Mask RCNN on a custom dataset with only 2 classes (3 if you count the background). However, any configuration I'm trying I don't seem to be able to get the rpn_bbox_loss lower than the value 2 (which I suppose has no real significance). In addition to this, the mrcnn_class_loss and mrcnn_mask_loss are always 0 (starting from epoch 1) and the mrcnn_bbox_loss is really low then gets to 0. The rpn_class_loss is the only one behaving normally (starting high then getting low).

My training images have the shapesroughly 200x200x3.
Should I change the IMAGE_MIN_DIM and IMAGE_MAX_DIM to 256 (this is how I've been trying to train with)?
If yes, what should my RPN_ANCHOR_SCALES be? They currently are (8, 16, 32, 64, 128)
Should I change any other config parameters for these shapes to work? I'm fairly new to training the mask rcnn so any help is more than welcome.
This is my full config:
Configurations:
BACKBONE                       resnet101
BACKBONE_STRIDES               [4, 8, 16, 32, 64]
BATCH_SIZE                     1
BBOX_STD_DEV                   [0.1 0.1 0.2 0.2]
COMPUTE_BACKBONE_SHAPE         None
DETECTION_MAX_INSTANCES        100
DETECTION_MIN_CONFIDENCE       0.8
DETECTION_NMS_THRESHOLD        0.3
EPOCHS                         25
FPN_CLASSIF_FC_LAYERS_SIZE     1024
GPU_COUNT                      1
GRADIENT_CLIP_NORM             5.0
IMAGES_PER_GPU                 1
IMAGE_CHANNEL_COUNT            3
IMAGE_MAX_DIM                  256
IMAGE_META_SIZE                15
IMAGE_MIN_DIM                  256
IMAGE_MIN_SCALE                0
IMAGE_RESIZE_MODE              square
IMAGE_SHAPE                    [256 256   3]
LEARNING_MOMENTUM              0.9
LEARNING_RATE                  0.001
LOSS_WEIGHTS                   {'rpn_class_loss': 1.0, 'rpn_bbox_loss': 1.0, 'mrcnn_class_loss': 3.0, 'mrcnn_bbox_loss': 1.0, 'mrcnn_mask_loss': 1.0}
MASK_POOL_SIZE                 14
MASK_SHAPE                     [28, 28]
MAX_GT_INSTANCES               100
MEAN_PIXEL                     [123.7 116.8 103.9]
MINI_MASK_SHAPE                (56, 56)
NAME                           animal
NUM_CLASSES                    3
POOL_SIZE                      7
POST_NMS_ROIS_INFERENCE        1000
POST_NMS_ROIS_TRAINING         2000
PRE_NMS_LIMIT                  6000
ROI_POSITIVE_RATIO             0.33
RPN_ANCHOR_RATIOS              [0.5, 1, 2]
RPN_ANCHOR_SCALES              (8, 16, 32, 64, 128,)
RPN_ANCHOR_STRIDE              1
RPN_BBOX_STD_DEV               [0.1 0.1 0.2 0.2]
RPN_NMS_THRESHOLD              0.7
RPN_TRAIN_ANCHORS_PER_IMAGE    256
STEPS_PER_EPOCH                20
TOP_DOWN_PYRAMID_SIZE          256
TRAIN_BN                       False
TRAIN_ROIS_PER_IMAGE           200
USE_MINI_MASK                  True
USE_RPN_ROIS                   True
VALIDATION_STEPS               1
WEIGHT_DECAY                   0.0001

Do any of these symptoms mean something to someone? Do you know what I might be doing wrong in my training? Is there a chance that I might have configured the Dataset class incorrectly?

![image](https://user-images.githubusercontent.com/33568833/119742072-a8693500-be8f-11eb-8714-0b6056c79f3c.png)
![image](https://user-images.githubusercontent.com/33568833/119742088-b323ca00-be8f-11eb-8d60-70776961a85b.png)
![image](https://user-images.githubusercontent.com/33568833/119742118-c20a7c80-be8f-11eb-825b-23583135e10e.png)
![image](https://user-images.githubusercontent.com/33568833/119742145-cc2c7b00-be8f-11eb-8d88-0f65c4f2bb5d.png)

When doing inference on the training images (just to prove that it doesn't work at all) the bounding boxes are totally messed up and so are the classes. You can also see that the mask is not consistent along the bbox, as it is somewhat "interrupted".
![image](https://user-images.githubusercontent.com/33568833/119742243-039b2780-be90-11eb-85b6-8b45611f2a33.png)

Thanks everyone!

**Comment by ashwin-999** [comment]
@SebastianNagy hi, what was the issue? both rpn, mrcnn bbox loss seems high for me. Did you figure out something to improve bbox loss?

**Comment by SebastianNagy** [comment]
> @SebastianNagy hi, what was the issue? both rpn, mrcnn bbox loss seems high for me. Did you figure out something to improve bbox loss?

Hi. I managed to fix it by using an older tag from the mask rcnn repository. I know it doesn't exactly make any sense but I changed model.py (leaving the other files from "mrcnn" folder as they are) with an older version and suddenly my training worked (quite well actually).

**Comment by SebastianNagy** [comment]
> @SebastianNagy hi, what was the issue? both rpn, mrcnn bbox loss seems high for me. Did you figure out something to improve bbox loss?

Leave me an email and I can send you tomorrow the exact mrcnn folder that I used, if you want :).

**Comment by AshwinAKannan** [comment]
> > @SebastianNagy hi, what was the issue? both rpn, mrcnn bbox loss seems high for me. Did you figure out something to improve bbox loss?
> 
> Leave me an email and I can send you tomorrow the exact mrcnn folder that I used, if you want :).

How do I email you?

**Comment by SebastianNagy** [comment]
> > > @SebastianNagy hi, what was the issue? both rpn, mrcnn bbox loss seems high for me. Did you figure out something to improve bbox loss?
> > 
> > 
> > Leave me an email and I can send you tomorrow the exact mrcnn folder that I used, if you want :).
> 
> How do I email you?
My bad for the ambiguous sentence. I meant to leave me your email :D.

**Comment by AshwinAKannan** [comment]
> > > > @SebastianNagy hi, what was the issue? both rpn, mrcnn bbox loss seems high for me. Did you figure out something to improve bbox loss?
> > > 
> > > 
> > > Leave me an email and I can send you tomorrow the exact mrcnn folder that I used, if you want :).
> > 
> > 
> > How do I email you?
> > My bad for the ambiguous sentence. I meant to leave me your email :D.

tempemailgithub@gmail.com. Thank you

**Comment by kecyarchana** [comment]
@SebastianNagy I am facing the same issue regarding bounding box loss (rpn_bbox_loss) I am using TensorFlow 2.0 can you please provide me the mrcnn folder you have used? My email is kecyarchana@gmail.com. Thanks

**Comment by f-macfarlane95** [comment]
I am having a similar issue - training on the balloon example results in high rpn_bbox_loss and mrcnn_bbox_loss loss and when investigating the rpn output it seems random. I am using tf 1.15, keras 2.1.6 and imgaug 0.4.0, I have tried the combinations that others report working but so far I have not found a stable combination.

![test](https://user-images.githubusercontent.com/6872236/155859750-76e70342-f0a8-4cab-8d9b-41d5edb8220a.png)
.

**Comment by jamajk** [comment]
Hi @kecyarchana I as well am using Tensorflow 2 for training, can you say if your training was fixed by using a different model.py?

**Comment by babanthierry94** [comment]
@kecyarchana  @SebastianNagy @jamajk 
Hi ;)
Please, i a facing the same issue can you please share with me your model.py.

**Comment by nyinyinyanlin** [comment]
@kecyarchana @SebastianNagy @jamajk @babanthierry94 I am having the same problem. Could you please kindly share me the exact older tag version or send the file to me? My email is nyinyinyanlin.mm@gmail.com

**Comment by chourib** [comment]
@nyinyinyanlin @kecyarchana @SebastianNagy @jamajk @babanthierry94 I am having the same problem. Could you please kindly share me the exact older tag version or send the file to me? My email is ikram@bpartners.app

**Comment by nyinyinyanlin** [comment]
@chourib Please refer to #2892 thread. I used Lee Kun Hee's fork and a set of specific versions for tf, keras and cuda which worked out for me.
